spec/parallel_tests/grouper_spec.rb

```diff
@@ -76,50 +76,116 @@ def call(num_groups, options = {})
+      end
 
-    it "specify_groups aborts when number of specs separated by pipe is out of bounds" do
-      expect do
-        call(3, specify_groups: '1|2|3|4')
-      end.to raise_error(
-        "Number of processes separated by pipe must be less than or equal to the total number of processes"
-      )
-    end
+      it "specify_groups aborts when number of specs separated by pipe is out of bounds" do
+        expect do
+          call(3, specify_groups: '1|2|3|4')
+        end.to raise_error(
+          "Number of processes separated by pipe must be less than or equal to the total number of processes"
+        )
+      end
 
-    it "specify_groups aborts when spec passed in doesn't match existing specs" do
-      expect do
-        call(3, specify_groups: '1|2|6')
-      end.to raise_error(
-        "Could not find [\"6\"] from --specify-groups in the selected files & folders"
-      )
-    end
+      it "specify_groups aborts when spec passed in doesn't match existing specs" do
+        expect do
+          call(3, specify_groups: '1|2|6')
+        end.to raise_error(
+          "Could not find [\"6\"] from --specify-groups in the selected files & folders"
+        )
+      end
 
-    it "specify_groups aborts when spec passed in doesn't match existing specs again" do
-      expect do
-        call(3, specify_groups: '1,6|2')
-      end.to raise_error(
-        "Could not find [\"6\"] from --specify-groups in the selected files & folders"
-      )
-    end
+      it "specify_groups aborts when spec passed in doesn't match existing specs again" do
+        expect do
+          call(3, specify_groups: '1,6|2')
+        end.to raise_error(
+          "Could not find [\"6\"] from --specify-groups in the selected files & folders"
+        )
+      end
 
-    it "specify_groups aborts when number of specs is equal to number passed in" do
-      expect do
-        call(3, specify_groups: '1|2|3')
-      end.to raise_error(/The specs that aren't run:\n\["4", "5"\]/)
+      it "specify_groups aborts when number of specs is equal to number passed in" do
+        expect do
+          call(3, specify_groups: '1|2|3')
+        end.to raise_error(/The specs that aren't run:\n\["4", "5"\]/)
+      end
+
+      it "specify_groups does not abort when the every single spec is specified in it" do
+        expect(call(3, specify_groups: '1,2|3,4|5')).to eq([["1", "2"], ["3", "4"], ["5"]])
+      end
     end
 
-    it "specify_groups does not abort when the every single spec is specified in it" do
-      expect(call(3, specify_groups: '1,2|3,4|5')).to eq([["1", "2"], ["3", "4"], ["5"]])
+    context 'with specific groups provided through STDIN' do
+      it "groups specify_groups as specified when specify_groups is just one spec" do
+        allow($stdin).to receive(:read).and_return("1\n")
+
+        expect(call(3, specify_groups: '-')).to eq([["1"], ["2", "5"], ["3", "4"]])
+      end
```

Comment:
don't think we need to copy-paste all the existing tests, 1 test to cover this edge-case should be enough ?

Reply:
I felt it would be good to cover the same input scenarios, to ensure it makes no difference whether data is directly provided or retrieved from stdin. But I am totally open to amend this if you feel strong.